iceoryx_posh/test/integrationtests/test_service_discovery.cpp

```diff
@@ -92,380 +71,355 @@ TEST_F(ServiceDiscovery_test, FindServiceWithWildcardsReturnsOnlyIntrospectionSe
     }
 }
 
-TEST_F(ServiceDiscovery_test, OfferSingleMethodServiceSingleInstance)
+TEST_F(ServiceDiscovery_test, FindServiceReturnsOfferedService)
 {
     ::testing::Test::RecordProperty("TEST_ID", "30f0e255-3584-4ab2-b7a6-85c16026852d");
-    auto isServiceOffered = sut.offerService({"service1", "instance1", "event1"});
-    this->InterOpWait();
+    const iox::capro::ServiceDescription SERVICE_DESCRIPTION("service", "instance", "event");
+    iox::popo::UntypedPublisher publisher(SERVICE_DESCRIPTION);
 
-    auto serviceContainer = sut.findService(IdString_t("service1"), IdString_t("instance1"));
+    auto serviceContainer =
+        sut.findService(SERVICE_DESCRIPTION.getServiceIDString(), SERVICE_DESCRIPTION.getInstanceIDString());
 
     ASSERT_FALSE(serviceContainer.has_error());
-    ASSERT_THAT(serviceContainer.value().size(), Eq(1u));
-    ASSERT_THAT(*serviceContainer.value().begin(), Eq(ServiceDescription{"service1", "instance1", "event1"}));
-    ASSERT_EQ(true, isServiceOffered);
-}
-
-
-TEST_F(ServiceDiscovery_test, OfferServiceWithDefaultServiceDescriptionFails)
-{
-    ::testing::Test::RecordProperty("TEST_ID", "1db1ce50-4e95-46f3-8682-9cc90576dbc0");
-    auto isServiceOffered = sut.offerService(iox::capro::ServiceDescription());
-    this->InterOpWait();
-
-    ASSERT_EQ(false, isServiceOffered);
+    ASSERT_THAT(serviceContainer.value().size(), Eq(1U));
+    EXPECT_THAT(*serviceContainer.value().begin(), Eq(SERVICE_DESCRIPTION));
 }
 
-TEST_F(ServiceDiscovery_test, OfferServiceWithValidEventIdSucessfull)
+TEST_F(ServiceDiscovery_test, ReofferedServiceWithValidServiceDescriptionCanBeFound)
 {
-    ::testing::Test::RecordProperty("TEST_ID", "1107d0e3-42e1-4b24-9a4d-cef8badb7154");
-    auto isServiceOffered = sut.offerService({"service1", "instance1", "event1"});
-    this->InterOpWait();
+    ::testing::Test::RecordProperty("TEST_ID", "b67b4990-e2fd-4efa-ab5d-e53c4ee55972");
+    const iox::capro::ServiceDescription SERVICE_DESCRIPTION("service", "instance", "event");
+    iox::popo::UntypedPublisher publisher(SERVICE_DESCRIPTION);
 
-    ASSERT_EQ(true, isServiceOffered);
-}
+    auto serviceContainer =
```

Comment:
The creation of the port updates the internal structures used in `findService` in a blocking way and hence the sleeps are not needed. Hence when we call `findService`, we can be sure the necessary updates already happened.  Is this correct @FerdinandSpitzschnueffler ?

This is important, otherwise we create spuriously failing tests (which the waits would mitigate to an extent).

Reply:
@MatthiasKillat Correct. The steps in code between calling the c'tor of the publisher and adding the service description to the registry are hard to explain without looking into the code simultaneously, but I could show you in a call if you like.